### PR TITLE
CIV-6180 claimant intend pa p set._Snyk

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-6180
https://tools.hmcts.net/jira/browse/CIV-6181
https://tools.hmcts.net/jira/browse/CIV-7059

### Change description ###

Claimant Intend: Response Task List - Part Admit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts [hunk 1]
```diff
@@ -2,15 +2,128 @@ import {t} from 'i18next';
...
+
+  const mediation = claim.claimantResponse?.mediation;
+
+  if (mediation?.mediationDisagreement?.option === YesNo.NO) {
+    freeTelephoneMediationTask.status = TaskStatus.COMPLETE;
+  } else {
+    if (mediation?.canWeUse?.option === YesNo.YES || mediation?.canWeUse?.mediationPhoneNumber) {
+      freeTelephoneMediationTask.status = TaskStatus.COMPLETE;
+    }
+    if (mediation?.companyTelephoneNumber?.option === YesNo.NO) {
```
Comment (laperla11) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts:62]: you can do `mediation?.companyTelephoneNumber?.option === YesNo.NO && hasClaimantResponseContactPersonAndCompanyPhone(claim)` here

**Review by laperla11 — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

File: src/main/common/form/models/claimantResponse/chooseHowToProceed.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {IsDefined} from 'class-validator';
+import {ChooseHowProceed} from '../../../models/chooseHowProceed';
```
Comment (portizdev) [inline src/main/common/form/models/claimantResponse/chooseHowToProceed.ts:2]: could we use in the shorter way those imports?

File: src/main/common/models/claim.ts [hunk 3]
```diff
@@ -442,6 +442,23 @@ export class Claim {
     }
     return party?.partyDetails?.partyName;
   }
+
+  get DQTrackType(): DQTrackOptions {
+    if (this.totalClaimAmount <= 10000) {
```
Comment (portizdev) [inline src/main/common/models/claim.ts:447]: Would it be better practice to set the values as global constants? I don't know if we need to use these values in more places, to facilitate maintainability.

File: src/main/routes/features/claimantResponse/chooseHowToProceedcController.ts [hunk 4]
```diff
@@ -1,12 +1,10 @@
 import {NextFunction, Request, Response, Router} from 'express';
 import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
-import {
-  CLAIMANT_RESPONSE_CHOOSE_HOW_TO_PROCEED_URL,
-  CLAIMANT_RESPONSE_TASK_LIST_URL,
-} from '../../urls';
+import {CLAIMANT_RESPONSE_CHOOSE_HOW_TO_PROCEED_URL, CLAIMANT_RESPONSE_TASK_LIST_URL,} from '../../urls';
```
Comment (portizdev) [inline src/main/routes/features/claimantResponse/chooseHowToProceedcController.ts:3]: check url import path

**Review by hmcts-version1-domini — DISMISSED** [review]

**Review by laperla11 — APPROVED** [review]

**Review by hmcts-version1-domini — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by GorilaNaranja — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Pull request merged**